distributed/deploy/tests/test_local.py

```diff
@@ -1155,3 +1155,22 @@ async def test_connect_to_closed_cluster():
         # Raises during init without actually connecting since we're not
         # awaiting anything
         Client(cluster, asynchronous=True)
+
+
+class MyPlugin:
+    def setup(self, worker=None):
+        import my_nonexistent_library  # noqa
+
+
+@gen_test(
+    clean_kwargs={
+        # FIXME: This doesn't close the LoopRunner properly, leaving a thread around
+        "threads": False
+    }
+)
+async def test_localcluster_start_exception():
+    with raises_with_cause(RuntimeError, None, ImportError, None):
+        async with LocalCluster(
+            plugins={MyPlugin()},
+        ):
+            return
```

Comment:
I added a simplified version of the LocalCluster test reproducer. The subclassing suggested in https://github.com/dask/distributed/pull/6363#issuecomment-1132821690 is raising an exception during init which avoids us going through proper cleanup by the contextmanager, i.e. resources are never closed. The interesting bit, however, is the plugin for the workers which can be just passed through as a kwarg

Reply:
cc @pentschev